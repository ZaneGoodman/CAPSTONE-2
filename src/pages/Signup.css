.Signup {
  text-align: center;
  margin: 40px;
}

.Signup-header {
  color: rgb(236, 234, 226);
  font-family: "Rosarivo", cursive;
  text-align: center;
}

.Signup-background {
  background-image: url("https://cdn.pixabay.com/photo/2014/10/22/18/16/church-498525_1280.jpg");

  background-size: cover;
  background-repeat: no-repeat;
  position: absolute;
  margin: 0;
  padding: 0;
  left: 0;
  top: 49px;
  bottom: 0;
  right: 0;
  overflow: hidden;
}
