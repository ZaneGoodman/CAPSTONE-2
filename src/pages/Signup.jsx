import React, { useState } from "react";
import { useAuth } from "../provider/authProvider";
import { useNavigate } from "react-router-dom";
import AuthForm from "../forms/AuthForm";
import Authorization from "../models/authUser";

const Signup = () => {
  const { setToken, setUsername } = useAuth();
  const navigate = useNavigate();

  const INITIAL_STATE = {
    username: "",
    password: "",
  };
  const [userData, setUserData] = useState(INITIAL_STATE);
  //Login - Check for error, respond with errors if present.
  //log in user, add token and username, navigate home page
  const handleLogin = async (fData) => {
    setUserData(() => fData);
    const response = await Authorization.signup(fData.username, fData.password)
      .then((res) => setToken(res))
      .catch(() => "invalid");

    if (response === "invalid") {
      return "invalid";
    } else {
      setUsername(fData.username);
      navigate("/", { replace: true });
      navigate(0);
    }
  };
  return (
    <>
      <h1>Sign Up</h1>
      <AuthForm login={handleLogin} />
    </>
  );
};

export default Signup;
