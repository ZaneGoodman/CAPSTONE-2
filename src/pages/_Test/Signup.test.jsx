import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import AuthProvider from "../../provider/authProvider";
import Signup from "../Signup";
//Smoke test
test("renders Signup component with authentication", async () => {
  render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <Signup />
      </MemoryRouter>
    </AuthProvider>
  );
});

//Snapshot test  <MemoryRouter>
it("Signup matches snapshot", async () => {
  const { asFragment } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <Signup />
      </MemoryRouter>
    </AuthProvider>
  );
  await expect(asFragment()).toMatchSnapshot();
});
