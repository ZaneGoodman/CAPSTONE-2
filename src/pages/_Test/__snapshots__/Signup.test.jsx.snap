// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Sign Up
  </h1>
  <form>
    <label
      for="username"
    >
      Username
    </label>
    <input
      autocomplete="username"
      id="username"
      name="username"
      placeholder="username"
      type="text"
      value=""
    />
    <label>
      Password
    </label>
    <input
      autocomplete="current-password"
      id="password"
      name="password"
      placeholder="password"
      type="password"
      value=""
    />
    <button>
      Go!
    </button>
  </form>
</DocumentFragment>
`;
